spec/services/process_interaction_service_spec.rb

```diff
@@ -31,6 +32,34 @@
     end
   end
 
+  describe 'remote follow request slap' do
+    before do
+      receiver.update(locked: true)
+      # Copy already-generated key
+      remote_sender.update(private_key: sender.private_key, public_key: remote_sender.public_key)
+
+      payload = <<XML
+<entry xmlns="http://www.w3.org/2005/Atom" xmlns:activity="http://activitystrea.ms/spec/1.0/">
+  <author>
+    <name>carol</name>
+    <uri>https://webdomain.com/users/carol</uri>
+  </author>
+
+  <id>someIdHere</id>
+  <activity:verb>http://activitystrea.ms/schema/1.0/request-friend</activity:verb>
+</entry>
+XML
+
+      envelope = OStatus2::Salmon.new.pack(payload, remote_sender.keypair)
+      subject.call(envelope, receiver)
```

Comment:
I just backported the tests I wrote for my previous pull request, but the discovery rules you used are a bit different. I'll just add the email tag to the request, as that is something we both decided to handle the same way.